dspace/modules/api/src/main/java/org/datadryad/rest/models/Author.java

```diff
@@ -52,31 +53,96 @@ public Author(String authorString) {
                     // throw this away.
                     authorString = namepattern.group(1);
                 } else {
-                    this.givenNames = namepattern.group(2);
-                    this.familyName = namepattern.group(1);
+                    setGivenNames(namepattern.group(2));
+                    setFamilyName(namepattern.group(1));
                     return;
                 }
             }
 
             // if it's firstname lastname
             namepattern = Pattern.compile("^(.+) +(.*)$").matcher(authorString);
             if (namepattern.find()) {
-                this.givenNames = namepattern.group(1);
-                this.familyName = namepattern.group(2) + suffix;
+                setGivenNames(namepattern.group(1));
+                setFamilyName(namepattern.group(2) + suffix);
             } else {
                 // there is only one word in the name: assign it to the familyName?
-                this.familyName = authorString;
-                this.givenNames = "";
+                setGivenNames("");
+                setFamilyName(authorString);
             }
         }
         return;
     }
 
-    public final String fullName() {
-        String name = familyName;
-        if (!"".equals(givenNames)) {
-            name = familyName + ", " + givenNames;
+    public void setFamilyName(String familyName) {
+        this.familyName = StringEscapeUtils.escapeHtml(familyName);
+    }
+
+    public void setGivenNames(String givenNames) {
+        this.givenNames = StringEscapeUtils.escapeHtml(givenNames);
+    }
+
+    public final String getUnicodeFullName() {
+        String name = getUnicodeFamilyName();
+        if (!"".equals(getUnicodeGivenNames())) {
+            name = getUnicodeFamilyName() + ", " + getUnicodeGivenNames();
         }
         return name;
     }
+
+    public String getUnicodeFamilyName() {
+        return StringEscapeUtils.unescapeHtml(familyName);
+    }
+
+    public String getUnicodeGivenNames() {
+        return StringEscapeUtils.unescapeHtml(givenNames);
+    }
+
+    public final String getHTMLFullName() {
+        String name = getHTMLFamilyName();
+        if (!"".equals(getHTMLGivenNames())) {
+            name = getHTMLFamilyName() + ", " + getHTMLGivenNames();
+        }
+        return name;
+    }
+
+    public String getHTMLFamilyName() {
```

Comment:
This is exactly the same function as getUnicodeFamilyName. Is there a need for both?